WordPress/Classes/Categories/NSString+Helpers.m

```diff
@@ -91,33 +91,46 @@ - (NSString *)stringByReplacingHTMLEmoticonsWithEmoji
-                                   @"smile": @"😊",
-                                   @"surprised": @"😮",
-                                   @"twisted": @"👿",
-                                   @"wink": @"😉"
+                                   @"icon_arrow": @"➡",
+                                   @"icon_biggrin": @"😃",
+                                   @"icon_confused": @"😕",
+                                   @"icon_cool": @"😎",
+                                   @"icon_cry": @"😭",
+                                   @"icon_eek": @"😮",
+                                   @"icon_evil": @"😈",
+                                   @"icon_exclaim": @"❗",
+                                   @"icon_idea": @"💡",
+                                   @"icon_lol": @"😄",
+                                   @"icon_mad": @"😠",
+                                   @"icon_mrgreen": @"🐸",
+                                   @"icon_neutral": @"😐",
+                                   @"icon_question": @"❓",
+                                   @"icon_razz": @"😛",
+                                   @"icon_redface": @"😊",
+                                   @"icon_rolleyes": @"😒",
+                                   @"icon_sad": @"😞",
+                                   @"icon_smile": @"😊",
+                                   @"icon_surprised": @"😮",
+                                   @"icon_twisted": @"👿",
+                                   @"icon_wink": @"😉",
+                                   // NOTE: There is not a perfect match for the following four with
+                                   // currently supported emoji. We'll try to get as close as we can.
+                                   @"frownie":@"😞",
+                                   @"mrgreen":@"😊",
+                                   @"rolleyes":@"😒",
+                                   @"simple-smile":@"😊"
                                    };
 
-    NSError *error = nil;
-    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@"<img src=['\"].*?wp-includes/images/smilies/icon_(.+?).gif['\"].*?class=['\"]wp-smiley['\"] ?/?>" options:NSRegularExpressionCaseInsensitive error:&error];
-    NSArray *matches = [regex matchesInString:result options:0 range:NSMakeRange(0, [result length])];
+    static NSRegularExpression *smiliesRegex;
+    static NSRegularExpression *coreEmojiRegex;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        NSError *error;
+        smiliesRegex = [NSRegularExpression regularExpressionWithPattern:@"<img.*?src=['\"].*?wp-includes/images/smilies/(.+?)(?:.gif|.png)[^'\"]*['\"].*?class=['\"]wp-smiley['\"].*?/?>" options:NSRegularExpressionCaseInsensitive error:&error];
+        coreEmojiRegex = [NSRegularExpression regularExpressionWithPattern:@"<img .*?src=['\"].*?images/core/emoji/[^/]+/(.+?).png['\"].*?class=['\"]wp-smiley['\"][^//]+/?>" options:NSRegularExpressionCaseInsensitive error:&error];
```

Comment:
There's actually a few emoji characters that don't match this pattern. For instance, from http://wpt.koke.me/2015/09/29/all-the-emoji/

```
$ curl -s wpt.koke.me/feed/ | grep -o '[0-9a-f-]*-[0-9a-f-]*.png" alt="[^"]*"' | sed -e 's/" alt="/ -> /'
```
> 1f1e6-1f1e7.png -> 🇦🇧"
1f1e8-1f1f3.png -> 🇨🇳"
1f1e8-1f1e9.png -> 🇨🇩"
1f1ea-1f1e9.png -> 🇪🇩"
1f1ea-1f1f8.png -> 🇪🇸"
1f1ea-1f1eb.png -> 🇪🇫"
1f1f7-1f1eb.png -> 🇷🇫"
1f1ec-1f1e7.png -> 🇬🇧"
1f1ec-1f1ed.png -> 🇬🇭"
1f1ee-1f1f9.png -> 🇮🇹"
1f1ee-1f1ef.png -> 🇮🇯"
1f1f5-1f1ef.png -> 🇵🇯"
1f1f0-1f1f7.png -> 🇰🇷"
1f1f0-1f1f1.png -> 🇰🇱"
1f1f2-1f1f3.png -> 🇲🇳"
1f1f4-1f1f5.png -> 🇴🇵"
1f1f6-1f1f7.png -> 🇶🇷"
1f1fa-1f1f7.png -> 🇺🇷"
1f1f8-1f1f9.png -> 🇸🇹"
1f1fa-1f1f8.png -> 🇺🇸"
1f1fa-1f1fb.png -> 🇺🇻"
1f1fc-1f1fd.png -> 🇼🇽"
1f1fe-1f1ff.png -> 🇾🇿"